### PR TITLE
Add additional tests for better coverage.

### PR DESCRIPTION
**Pull request by JoshDunnDev**
- Increases coverage to 96.47%.
- Codecov: https://app.codecov.io/gh/digitalbazaar/bedrock-validation/branch/update-test-coverage

### CODE REVIEW

File: test/mocha/001-schemas.js [hunk 1]
```diff
@@ -83,6 +91,268 @@ describe('bedrock-validation', function() {
         should.not.exist(result);
         err.name.should.equal('UnknownSchema');
       });
+    it('should return middleware with a ValidationError due to invalid body',
+      function() {
```
Comment (dlongley) [inline test/mocha/001-schemas.js:95]: We should pass `done` in here as a callback and then call it in `next` below. The reason for doing this is that `next` could be called on a different turn of the event loop. It's true that it presently *isn't*, but better to model good behavior in our tests if we can.

```suggestion
      function(done) {
```

File: test/mocha/001-schemas.js [hunk 2]
```diff
@@ -83,6 +91,268 @@ describe('bedrock-validation', function() {
...
+        const req = {
+          body: ''
+        };
+        const res = {};
+        const next = function(err) {
+          should.exist(err);
+          err.name.should.equal('ValidationError');
+        };
```
Comment (dlongley) [inline test/mocha/001-schemas.js:103]:
```suggestion
          done();
        };
```

File: test/mocha/001-schemas.js [hunk 3]
```diff
@@ -83,6 +91,268 @@ describe('bedrock-validation', function() {
...
+      result.valid.should.be.true;
+    });
+  });
+
+  describe('w3cDateTime', function() {
+    const schema = validation.getSchema('w3cDateTime');
+    it('should be an Object', function() {
+      schema.should.be.an.instanceof(Object);
+    });
+    it('should accecpt valid w3cDateTime', function() {
```
Comment (dlongley) [inline test/mocha/001-schemas.js:266]:
```suggestion
    it('should accept valid w3cDateTime', function() {
```

File: test/mocha/001-schemas.js [hunk 4]
```diff
@@ -83,6 +91,268 @@ describe('bedrock-validation', function() {
...
+      result.valid.should.be.true;
+    });
+  });
+
+  describe('url', function() {
+    const schema = validation.getSchema('url');
+    it('should be an Object', function() {
+      schema.should.be.an.instanceof(Object);
+    });
+    it('should accecpt valid url', function() {
```
Comment (dlongley) [inline test/mocha/001-schemas.js:243]:
```suggestion
    it('should accept valid url', function() {
```

File: test/mocha/001-schemas.js [hunk 5]
```diff
@@ -83,6 +91,268 @@ describe('bedrock-validation', function() {
...
+      result.valid.should.be.true;
+    });
+  });
+
+  describe('personName', function() {
+    const schema = validation.getSchema('personName');
+    it('should be an Object', function() {
+      schema.should.be.an.instanceof(Object);
+    });
+    it('should accecpt valid personName', function() {
```
Comment (dlongley) [inline test/mocha/001-schemas.js:290]: Search and replace `accecpt` typo.

File: test/mocha/001-schemas.js [hunk 6]
```diff
@@ -83,6 +91,268 @@ describe('bedrock-validation', function() {
...
+      result.valid.should.be.true;
+    });
+    it('should reject an invalid personName', function() {
+      const result = validation.validate('personName', {});
+      result.valid.should.be.false;
+    });
+    it('should accept valid personName with extend', function() {
+      const extend = {name: 'test'};
+      // eslint-disable-next-line max-len
+      const schema = require('../node_modules/bedrock-validation/schemas/personName')(extend);
```
Comment (dlongley) [inline test/mocha/001-schemas.js:301]: Not sure if this approach is going to survive moving to ESM. Is there another way? If not -- ok for now, we'll deal with ESM later.

**Review by dlongley — APPROVED** [review]
Approving, but see minor comments to address.

File: test/mocha/001-schemas.js (hunk 6 above)
Comment (JoshDunnDev) [reply test/mocha/001-schemas.js:301]: Well there probably is but there were also many tests already in this library which is why I used this structure. I think it's best we change that when we work on the ESM conversion.

**Pull request merged**